subsys/bluetooth/host/att.c

```diff
@@ -607,14 +606,27 @@ static u8_t find_type_cb(const struct bt_gatt_attr *attr, void *user_data)
 		 * Since we don't know if it is the service with requested UUID,
 		 * we cannot respond with an error to this request.
 		 */
-		data->group = NULL;
-		return BT_GATT_ITER_CONTINUE;
+		goto uuid_iterate_skip;
 	}
 
 	/* Check if data matches */
-	if (read != data->value_len || memcmp(data->value, uuid, read)) {
-		data->group = NULL;
-		return BT_GATT_ITER_CONTINUE;
+	if (read != data->value_len) {
+		/* Use bt_uuid_cmp() to compare UUIDs of different form. */
+		struct bt_uuid_128 ref_uuid;
+		struct bt_uuid_128 recvd_uuid;
+
+		if (!bt_uuid_create_le(&recvd_uuid.uuid, data->value,
+				       data->value_len)) {
+			goto uuid_iterate_skip;
+		}
+		if (!bt_uuid_create(&ref_uuid.uuid, uuid, read)) {
+			goto uuid_iterate_skip;
```

Comment:
Perhaps we should log something if we fail to create the UUID, i.e. BT_WARN("Unable to create UUID: size %d", read)...